File: healthtrack/test_notifications_fix.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

Future<void> main() async {
  // Test the notifications endpoint
  final baseUrl = 'http://localhost:3000';
  final endpoint = '$baseUrl/admin/notifications';
  
  print('Testing notifications endpoint: $endpoint');
  
  try {
    final response = await http.get(
      Uri.parse(endpoint),
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
      },
    );
    
    print('Status code: ${response.statusCode}');
    print('Content-Type: ${response.headers['content-type']}');
    
    if (response.headers['content-type']?.contains('text/html') ?? false) {
      print('ERROR: Server returned HTML instead of JSON');
      print('Response body: ${response.body}');
    } else {
      print('SUCCESS: Server returned JSON');
      final data = json.decode(response.body);
      print('Response data: $data');
    }
  } catch (e) {
    print('ERROR: Failed to connect to server: $e');
  }
}

File: healthtrack/test_notifications_fix.js
const http = require('http');

// Test the notifications endpoint
const options = {
  hostname: 'localhost',
  port: 3000,
  path: '/admin/notifications',
  method: 'GET',
  headers: {
    'Content-Type': 'application/json',
    'Accept': 'application/json'
  }
};

console.log('Testing notifications endpoint: http://localhost:3000/admin/notifications');

const req = http.request(options, (res) => {
  console.log('Status code:', res.statusCode);
  console.log('Content-Type:', res.headers['content-type']);
  
  let data = '';
  
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    if (res.headers['content-type'] && res.headers['content-type'].includes('text/html')) {
      console.log('ERROR: Server returned HTML instead of JSON');
      console.log('Response body:', data);
    } else {
      console.log('SUCCESS: Server returned JSON');
      try {
        const jsonData = JSON.parse(data);
        console.log('Response data:', JSON.stringify(jsonData, null, 2));
      } catch (e) {
        console.log('Response body:', data);
      }
    }
  });
});

req.on('error', (e) => {
  console.error('ERROR: Failed to connect to server:', e.message);
});

req.end();